common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -42,77 +45,100 @@
  *
  * @author slievrly
  */
-public class EnhancedServiceLoader {
+public class EnhancedServiceLoader<S> {
     private static final Logger LOGGER = LoggerFactory.getLogger(EnhancedServiceLoader.class);
     private static final String SERVICES_DIRECTORY = "META-INF/services/";
     private static final String SEATA_DIRECTORY = "META-INF/seata/";
-    @SuppressWarnings("rawtypes")
-    private static Map<Class, List<Class>> providers = new ConcurrentHashMap<>();
+
+    private static final ConcurrentMap<Class<?>, EnhancedServiceLoader<?>> SERVICE_LOADERS =
+            new ConcurrentHashMap<>();
+    private static final ConcurrentMap<Class<?>, Object> EXTENSION_INSTANCES = new ConcurrentHashMap<>();
+
+    private final Class<S> type;
+    private final Holder<Map<ExtensionURL, Class<?>>> extensionClasses = new Holder<>();
+    private final ConcurrentMap<ExtensionURL, Holder<Object>> extensionInstances = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String, List<ExtensionURL>> extensionNameUrlsMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Class<?>, ExtensionURL> extensionClassUrlMap = new ConcurrentHashMap<>();
+    private ExtensionURL defaultExtensionURL = null;
+    private Integer highestLoadPriority = Integer.MIN_VALUE;
+
+    private EnhancedServiceLoader(Class<S> type) {
+        this.type = type;
+    }
+
+    /**
+     * Get the ServiceLoader for the specified Class
+     *
+     * @param type the type of the extension point
+     * @param <S> the type
+     * @return the service loader
+     */
+    public static <S> EnhancedServiceLoader<S> getServiceLoader(Class<S> type) {
```

Comment:
Keep the EnhancedServiceLoader's api unchanged?

Reply:
OK，I've restore the old api